schemas/client.yml

```diff
@@ -59,34 +59,91 @@
       description: >
         Client domain.
 
+    - name: name
+      level: core
+      type: keyword
+      short: Name of the client system.
+      description: >
+        Name of the client system.
+
+        It can contain what `hostname` returns on Unix systems, the fully
+        qualified domain name, or a name specified by the user. The sender
+        decides which value to use.
+        
+        In the case of fully and partially qualified domain names it is recommended to parse out 
+        the additional fields of top_level_domain, registered_domain, subdomain, and hostname
+        as they are available.
```

Comment:
How would whether this is a domain name or a custom name be determined? By whether the name contains a dot?

Perhaps "name" should work like "address" and you copy the best available name from more specific fields, or have a custom user supplied one? It feels somewhat overloaded here, it's a shame domain is used for other purposes in some field set groups.

Reply:
This one comes down to processing  - I'm honestly not sure what the results would look like for a partial domain in the logstash tld filter (will test and see)